### ISSUE TITLE
Training Failed with a MultiDiscrete Action Space Custom Environment

### ISSUE THREAD

**Issue by changxin-ge** [ray-project/ray#33007]
## Problem
Training failed immediately with a custom environment that contains MultiDiscrete action space. 

## Background
The error is related to a calculation in a `logp` operation where the label is outside the valid range (eg. label 101 is outside the valid range of [0, 101).). The range is consistent with one of the specifications of the MultiDiscrete action space (MultiDiscrete([20, 20, 5, 5, 101]). I also tried to use Tuple([Discrete, Discrete, ...]), and the same error came up. I also tried a number of different algorithms (PPO, APPO, etc.), and all failed in the same way. 

## Expected Result
Expecting algorithm training to work with MultiDiscrete action space. If the library can't be updated at the moment, please suggest a temporary solution. Thank you!

## Error Message
> Detected at node 'default_policy_wk1/SparseSoftmaxCrossEntropyWithLogits_9/SparseSoftmaxCrossEntropyWithLogits' defined at (most recent call last):
    File "/Users/px/miniforge3/lib/python3.10/site-packages/ray/_private/workers/default_worker.py", line 226, in <module>
      ray._private.worker.global_worker.main_loop()
    File "/Users/px/miniforge3/lib/python3.10/site-packages/ray/_private/worker.py", line 772, in main_loop
      self.core_worker.run_task_loop()
    File "/Users/px/miniforge3/lib/python3.10/site-packages/ray/_private/function_manager.py", line 674, in actor_method_executor
      return method(__ray_actor, *args, **kwargs)
    File "/Users/px/miniforge3/lib/python3.10/site-packages/ray/util/tracing/tracing_helper.py", line 466, in _resume_span
      return method(self, *_args, **_kwargs)
    File "/Users/px/miniforge3/lib/python3.10/site-packages/ray/rllib/evaluation/rollout_worker.py", line 737, in __init__
      self._build_policy_map(policy_dict=self.policy_dict)
    File "/Users/px/miniforge3/lib/python3.10/site-packages/ray/util/tracing/tracing_helper.py", line 466, in _resume_span
      return method(self, *_args, **_kwargs)
    File "/Users/px/miniforge3/lib/python3.10/site-packages/ray/rllib/evaluation/rollout_worker.py", line 1984, in _build_policy_map
      new_policy = create_policy_for_framework(
    File "/Users/px/miniforge3/lib/python3.10/site-packages/ray/rllib/utils/policy.py", line 130, in create_policy_for_framework
      return policy_class(
    File "/Users/px/miniforge3/lib/python3.10/site-packages/ray/rllib/algorithms/ppo/ppo_tf_policy.py", line 83, in __init__
      base.__init__(
    File "/Users/px/miniforge3/lib/python3.10/site-packages/ray/rllib/policy/dynamic_tf_policy_v2.py", line 95, in __init__
      ) = self._init_action_fetches(timestep, explore)
    File "/Users/px/miniforge3/lib/python3.10/site-packages/ray/rllib/policy/dynamic_tf_policy_v2.py", line 618, in _init_action_fetches
      action_dist = self.dist_class(dist_inputs, self.model)
    File "/Users/px/miniforge3/lib/python3.10/site-packages/ray/rllib/models/tf/tf_action_dist.py", line 134, in __init__
      self.sampled_action_logp_op = self.logp(self.sample_op)
    File "/Users/px/miniforge3/lib/python3.10/site-packages/ray/rllib/models/tf/tf_action_dist.py", line 157, in logp
      logps = tf.stack([cat.logp(act) for cat, act in zip(self.cats, actions)])
    File "/Users/px/miniforge3/lib/python3.10/site-packages/ray/rllib/models/tf/tf_action_dist.py", line 157, in <listcomp>
      logps = tf.stack([cat.logp(act) for cat, act in zip(self.cats, actions)])
    File "/Users/px/miniforge3/lib/python3.10/site-packages/ray/rllib/models/tf/tf_action_dist.py", line 67, in logp
      return -tf.nn.sparse_softmax_cross_entropy_with_logits(
Node: 'default_policy_wk1/SparseSoftmaxCrossEntropyWithLogits_9/SparseSoftmaxCrossEntropyWithLogits'
Received a label value of 101 which is outside the valid range of [0, 101).  Label values: 101
	 [[{{node default_policy_wk1/SparseSoftmaxCrossEntropyWithLogits_9/SparseSoftmaxCrossEntropyWithLogits}}]]

**Comment by changxin-ge** [comment]
Closing the Issue. 

I found the bug in my own code, the problem was caused by providing a numpy array that contains np.int64 and np.inf values and it was cast into np.float32. 

## Example of My Bug
```python
not_available = np.inf
arr1 = np.hstack([np.arrary([1, 2, 3]).astype(np.int64), not_available]).astype(np.float32)
arr2 = np.full(6, not_available).astype(np.float32)
arr3 = np.hstack([arr1, arr2])

# then arr3 was passed to observation
observation_space = Dict({
    ...,
    "key": Box(0, np.inf, (10,), np.float32)
})

obs = {
    ...,
    "key": arr3
}
```

## Solution
Use all np.int64 number in the numpy array and change the box limit. Use `-1` to indicate `not_available`. 
```python
not_available = -1
arr1 = np.hstack([np.arrary([1, 2, 3]), not_available]).astype(np.int64)
arr2 = np.full(6, not_available).astype(np.int64)
arr3 = np.hstack([arr1, arr2])

# then arr3 was passed to observation
observation_space = Dict({
    ...,
    "key": Box(-1, np.inf, (10,), np.int64)
})

obs = {
    ...,
    "key": arr3
}
```
